ballerina/tests/22_variables.bal

```diff
@@ -359,3 +359,59 @@ isolated function testScalarTypeVariableWithInputObjectValue() returns error? {
     json expectedPayload = check getJsonContentFromFile("scalar_type_variable_with_input_object_value.json");
     assertJsonValuesWithOrder(actualPayload, expectedPayload);
 }
+
+@test:Config {
+    groups: ["variables", "inputs", "enums"]
+}
+isolated function testEnumTypeDefaultValueWithVariables() returns error? {
+    string document = "($day:Weekday = SUNDAY){ isHoliday(weekday: $day) }";
+    string url = "http://localhost:9091/inputs";
+    json actualPayload = check getJsonPayloadFromService(url, document);
+    json expectedPayload = {
+        data: {
+            isHoliday: true
+        }
+    };
+    assertJsonValuesWithOrder(actualPayload, expectedPayload);
+}
+
+@test:Config {
+    groups: ["variables", "inputs", "enums"]
+}
+isolated function testInvalidEnumTypeDefaultValueWithVariables() returns error? {
```

Comment:
Shall we add a test for passing a string literal for a enum input, and the other way around as well?

Reply:
Currently, GraphQL `string` type variables accept `enum` values as the default value, and other way around as well. This is tracked via a separate issue [#1990](https://github.com/ballerina-platform/ballerina-standard-library/issues/1990)